perf/benchmark/graph_plotter/graph_plotter.py

```diff
@@ -0,0 +1,184 @@
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+metric_dict = {"cpu-client": "cpu_mili_avg_istio_proxy_fortioclient",
+               "cpu-server": "cpu_mili_avg_istio_proxy_fortioserver",
+               "mem-client": "mem_Mi_avg_istio_proxy_fortioclient",
+               "mem-server": "mem_Mi_avg_istio_proxy_fortioserver"}
+
+
+def plotter(args):
+    check_if_args_provided(args)
+
+    df = pd.read_csv(args.csv_filepath)
+    telemetry_modes_y_data = {}
+    metric_name = get_metric_name(args)
+    constructed_query_str = get_constructed_query_str(args)
+
+    for telemetry_mode in args.telemetry_modes:
+        telemetry_modes_y_data[telemetry_mode] = get_data_helper(df, args.query_list, constructed_query_str,
+                                                                 telemetry_mode, metric_name)
+
+    dpi = 100
+    plt.figure(figsize=(1138 / dpi, 871 / dpi), dpi=dpi)
+    for key, val in telemetry_modes_y_data.items():
+        plt.plot(args.query_list, val, marker='o', label=key)
+
+    plt.xlabel(get_x_label(args))
+    plt.ylabel(get_y_label(args))
+    plt.legend()
+    plt.grid()
+    plt.savefig(args.graph_title, dpi=dpi)
+    plt.show()
+
+
+# Helpers
+def check_if_args_provided(args):
+    args_all_provided = True
+    # print(vars(args))
+    for _, val in vars(args).items():
+        if val == "":
+            print("Warning: There is at least one argument that you did not specify with a value.\n")
+            args_all_provided = False
+    if not check_args_consistency(args):
+        args_all_provided = False
+    if not args_all_provided:
+        sys.exit(-1)
+
+
+def check_args_consistency(args):
+    if args.x_axis == "conn" and not args.query_str.startswith("ActualQPS=="):
+        print("Warning: your specified query_str is not match with the x_axis definition.")
```

Comment:
nit: these should be "does not match"

Reply:
will make this change in my next PR, thanks!